mlflow/metrics/genai/genai_metric.py

```diff
@@ -64,18 +64,18 @@ def _extract_score_and_justification(output):
             data = json.loads(text)
             score = int(data.get("score"))
             justification = data.get("justification")
-        except json.JSONDecodeError:
+        except (json.JSONDecodeError, ValueError) as e:
             # If parsing fails, use regex
             match = re.search(r"score: (\d+),?\s*justification: (.+)", text)
             if match:
                 score = int(match.group(1))
                 justification = match.group(2)
             else:
                 score = None
-                justification = None
+                justification = f"Failed to extract score and justification. Error: {e!s}"
```

Comment:
I think we should add the raw output here. The error message is not super helpful IMO. 
`Error: Expecting value: line 1 column 1 (char 0)"` 

Failed to extract score and justification. Here is the raw output: {output}

Something like this would be better. What do you think?